src/main/java/io/burt/athena/result/s3/InputStreamResponseTransformer.java

```diff
@@ -127,17 +129,22 @@ public int available() throws IOException {
         }
     }
 
-    private boolean ensureChunk() throws IOException {
-        if (error != null) {
-            throw new IOException(error);
-        }
+    protected boolean ensureChunk() throws IOException {
         if (readChunk == END_MARKER) {
+            if (error != null) {
+                throw new IOException(error);
+            }
             return false;
         } else if (readChunk == null || !readChunk.hasRemaining()) {
             try {
-                readChunk = chunks.take();
+                readChunk = chunks.poll(timeout.toMillis(), TimeUnit.MILLISECONDS);
                 if (readChunk == END_MARKER) {
+                    if (error != null) {
+                        throw new IOException(error);
+                    }
                     return false;
+                } else if (readChunk == null) {
+                    throw new IOException("read timeout");
```

Comment:
I don't know of a standard set of messages, but a quick google for stack traces seems to indicated that "Stream closed" and "Socket closed" as typical closed stream errors, and "Read timed out" seems to be "popular" with read timeouts. 

> Could this error be more informative? Like "Timed out while reading S3 response" (or even including the S3 URI, since we have the request object in scope).

It could, and I'm not really opposed to a longer message, but I'm not sure in which situation it would help. If we are targeting software, we would probably rather want to throw a more specific error type, if we are targeting developers debugging, then the stack trace should provide ample information, and if we are targeting end users, I'm not sure the message will make a difference.